library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java

```diff
@@ -34,6 +34,7 @@
     private static final String[] PROJECT_BATCH_ID = {DownloadContract.Batches._ID};
     private static final String WHERE_DELETED_VALUE_IS = DownloadContract.Batches.COLUMN_DELETED + " = ?";
     private static final String[] MARKED_FOR_DELETION = {"1"};
+    private static final String SORT_BY_ID_DESCENDING = DownloadContract.Batches._ID + " DESC";
```

Comment:
yep so by adding a new download to the queue, it will trigger the check on all previously queued items, and if you have a queue of 100 for example, and all of them fail but the newly added item, there will be 100 errors displayed